src/app/modules/shared_models/currency_amount.nim

```diff
@@ -0,0 +1,63 @@
+import NimQml, strformat
+
+QtObject:
+  type CurrencyAmount* = ref object of QObject
+    amount: float64
+    symbol: string
+    displayDecimals: int
+    stripTrailingZeroes: bool
+
+  proc setup(self: CurrencyAmount) =
+    self.QObject.setup
+
+  proc delete*(self: CurrencyAmount) =
+    self.QObject.delete
+
+  proc newCurrencyAmount*(
+    amount: float64,
+    symbol: string,
+    displayDecimals: int,
+    stripTrailingZeroes: bool,
+  ): CurrencyAmount =
+    new(result, delete)
+    result.setup
```

Comment:
You have to call setup after initializing all the members. Probably this is the reason for your broken properties.

Reply:
Hmm interesting... I'll try this now, but I'd like to get a better idea of why I'm doing this.
I understand the call to `QObject.setup` as the "construction of the base class", thinking in C++ terms, so the logical thing to do for me would be to do that first before doing more stuff with the other members. Should I think about this some other way?

Making a quick search across the codebase shows that `result.QObject.setup` is often called first, though there are a couple of cases where it is called at the end.

Reply:
Didn't work :(

Reply:
Thanks for the clarifications. Moving setup after the properties fixed some issues exposing `QObject`s through properties that were not being correctly deleted. I assumed it would work for you, I guess :).